src/cargo/ops/registry.rs

```diff
@@ -310,6 +365,10 @@ pub fn configure_http_handle(config: &Config, handle: &mut Easy) -> CargoResult<
     handle.low_speed_limit(10 /* bytes per second */)?;
     handle.low_speed_time(Duration::new(30, 0))?;
     handle.useragent(&version().to_string())?;
+    // Not all cURL builds support HTTP/2, ignore any errors.
+    if config.get_bool("http.http2")?.map(|x| x.val).unwrap_or(true) {
+        let _ = handle.http_version(HttpVersion::V2TLS);
```

Comment:
Should this perhaps emit a warning if an error happens?

Reply:
We have this option turned on by default, so it could be annoying and confusing to those using old libcurl.

Reply:
Oh good point yeah, I was thinking that if you explicitly configured `http2 = true` and it wasn't available we may want to warn, but it looks like this only exists for `http2 = false` so I think it's ok to not warn here